airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -318,18 +318,18 @@ public List<CheckedConsumer<JdbcDatabase, Exception>> getCheckOperations(final J
         PostgresUtils.checkFirstRecordWaitTime(config);
       });
 
-      // Verify that the db user has required privilege to perform replication.
+      // Verify that the db user has required privilege to perform replication. The user has to be super user of to has replication access.
       checkOperations.add(database -> {
         final String userName = config.get("username").asText();
 
         final List<JsonNode> userPrivileges = database.queryJsons(connection -> {
-          final PreparedStatement ps = connection.prepareStatement("SELECT userepl FROM pg_user WHERE usename = ?");
+          final PreparedStatement ps = connection.prepareStatement("SELECT userepl or usesuper AS has_access FROM pg_user WHERE usename = ?");
           ps.setString(1, userName);
           LOGGER.info("Verifying required privileges for user: {}", userName);
           return ps;
         }, sourceOperations::rowToJson);
 
-        if (!userPrivileges.get(0).get("userepl").asBoolean()) {
+        if (!userPrivileges.get(0).get("has_access").asBoolean()) {
```

Comment:
Do we know whether the cloud users that had the problem with 1.0.45 will have this permission?

Reply:
I have asked to the user to understand how they set up. Based on that I will adapt the PR, but this should be enough (supposedly)